pkg/apis/mysql/v1/cluster_test.go

```diff
@@ -84,3 +85,20 @@ func TestRequiresConfigMount(t *testing.T) {
 		t.Errorf("Cluster with configRef should require a config mount")
 	}
 }
+
+func TestRequiresCustomSSLSetup(t *testing.T) {
+	cluster := &MySQLCluster{}
```

Comment:
Not relevant to this PR but I'd be in favour of abstracting this given how often it comes up in the codebase and how easy it would be to make a mistake by not setting defaults. 

cluster := NewMySQLCluster() | NewMySQLClusterWithDefaults()